File: JavaScript/challenge_4.js

function isPrime(n) {
  var i = 2
    , sqrt_n = Math.floor(Math.sqrt(n))
    ;
  while (i <= sqrt_n) {
    if (n % i === 0 ) {
      return false;
    }
    i += 1;
  }
  return true;
}

function sumPrimes(maxPrimes, showLog) {
  var primeCount = 0
    , primeSum = 0
    , iValue = 2
    ;
  while (primeCount < maxPrimes) {
    if (isPrime(iValue) ) {
      primeCount += 1;
      primeSum += iValue;
    }
    iValue += 1;
  }
  return primeSum;
}

console.log(sumPrimes(1000));

File: Ruby/challenge_4.rb
# sum of first 1000 primes
# output: 3_682_913


def is_prime x
  (2..Math.sqrt(x)).none? {|f| x % f == 0}
end

max_primes = ARGV[0].to_i || 1000
prime_count = 0
prime_sum = 0
int_value = 2
primes = []
while prime_count < (max_primes) do
  if is_prime(int_value)
    primes << int_value
    prime_count += 1
    prime_sum += int_value
  end
  int_value += 1
end
#puts "Found #{prime_count} primes that sum to #{prime_sum}"
puts prime_sum
puts primes.join(' ')


